### PR TITLE
Lock reports and retry db connect (merge 1st)

### PR DESCRIPTION
**Pull request by curusarn**

### CODE REVIEW

File: server/handlers/game.go [hunk 1]
```diff
@@ -2,10 +2,11 @@ package handlers
 
 import (
 	"encoding/json"
-	"github.com/rs/zerolog/log"
 	"io/ioutil"
 	"net/http"
 
+	"github.com/rs/zerolog/log"
+
```
Comment (curusarn) [inline server/handlers/game.go:9]: my editor prefers it this way I guess

File: server/handlers/game.go [hunk 2]
```diff
@@ -107,7 +108,35 @@ func Commands(game *stycoon.Game, w http.ResponseWriter, req *http.Request) {
...
+	}
+	// TODO: maybe only for logged users?
+	// var err error
+	// _, err = stycoon.LoggedUserFromSession(req, game.SessionManager)
+	// if err != nil {
+	// 	log.Warn().Err(err).Msg("User is not logged in")
+	// 	http.Error(w, "only for logged users", http.StatusForbidden)
+	// 	return
+	// }
+	// TODO: maybe only return player reports?
```
Comment (curusarn) [inline server/handlers/game.go:130]: Visualization was getting 403 when I required login for reports. So I just commented it out. Idk if we want all reports to be public or if we want to filter them by player.

File: server/handlers/game.go (hunk 1 above)
Comment (tivvit) [reply server/handlers/game.go:9]: isn't this part of go fmt?

File: server/handlers/game.go (hunk 2 above)
Comment (tivvit) [reply server/handlers/game.go:130]: you are right it should be public

File: server/handlers/game.go (hunk 1 above)
Comment (curusarn) [reply server/handlers/game.go:9]: I think goimports did the reordering

File: server/database/db.go [hunk 3]
```diff
@@ -24,18 +41,32 @@ func ConnectDB() *sql.DB {
...
-	db, err := sql.Open("mysql", cfg.FormatDSN())
-	if err != nil {
-		log.Fatal().Err(err).Msg("DB connection failed")
+	retry := 0
+	var db *sql.DB
+	for {
+		var err error
+		db, err = ConnectDB(cfg)
+		if err != nil {
+			log2 := log.With().Err(err).Int("retry", retry).Int("max_retries", max_retries).Logger()
```
[the remark this answers is not in the document]
Comment (curusarn) [reply server/database/db.go]: renamed

File: server/database/db.go [hunk 4]
```diff
@@ -24,18 +41,32 @@ func ConnectDB() *sql.DB {
...
+			if retry < max_retries {
+				log2.Error().Msg("DB connection failed - retrying")
+				retry++
+				time.Sleep(retry_backoff)
```
[the remark this answers is not in the document]
Comment (curusarn) [reply server/database/db.go:54]: Increased both max retry count (to 5) and retry backoff (to 500ms)

**Pull request merged**